packages/sources/moonbeam-address-list/src/endpoint/address.ts

```diff
@@ -0,0 +1,130 @@
+import { AdapterEndpoint } from '@chainlink/external-adapter-framework/adapter'
+import { ResponseCache } from '@chainlink/external-adapter-framework/cache/response'
+import { AdapterConfig } from '@chainlink/external-adapter-framework/config'
+import { Transport, TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { AdapterRequest, AdapterResponse } from '@chainlink/external-adapter-framework/util'
+import { Cache } from '@chainlink/external-adapter-framework/cache'
+import { ethers } from 'ethers'
+import { customSettings } from '../config'
+import { MoonbeamAddressContract_ABI } from '../abi/MoonbeamAddressContractABI'
+import { encodeAddress } from '@polkadot/keyring'
+
+type NetworkChainMap = { [network: string]: { [chain: string]: string } }
+
+const networks = ['moonbeam']
+const chainIds = ['mainnet', 'testnet']
+
+const networkChainMap: NetworkChainMap = {
+  moonbeam: {
+    mainnet: '0xFA36Fe1dA08C89eC72Ea1F0143a35bFd5DAea108',
+    testnet: '',
+  },
+}
+
+const inputParameters = {
+  contractAddress: {
+    description: 'The address of the Address Manager contract holding the custodial addresses.',
+    type: 'string',
+  },
+  chainId: {
+    description: 'The name of the target custodial chain',
+    options: chainIds,
+    type: 'string',
+    default: 'mainnet',
+  },
+  network: {
+    description: 'The name of the target custodial network protocol',
+    options: networks,
+    type: 'string',
+    default: 'moonbeam',
+  },
+} as const
+
+interface PorInputAddress {
+  network: string
+  chainId: string
+  address: string
+}
+
+interface RequestParams {
+  contractAddress: string
+  network: string
+  chainId: string
+}
+
+interface ResponseSchema {
+  Data: {
+    result: PorInputAddress[]
+  }
+  Result: null
+}
```

Comment:
In theory a lot of this is standardized on our end for PoR feeds, what do you think of adding these as classes in the framework? We can either do it now and push this EA using those, or we can refactor this EA after we change the fw

Reply:
Let's handle this later. I'll come back and give these adapters a bump after